src/Azure.Functions.Cli/Helpers/AzureHelper.cs

```diff
@@ -357,9 +390,75 @@ private static async Task ArmHttpAsync(HttpMethod method, Uri uri, string access
         {
-            var functions = await GetFunctions(functionApp, accessToken, managementURL);
+            var masterKey = await GetMasterKeyAsync(functionApp.SiteId, accessToken, managementURL);
+
+            if (functionAppReadyClient == null)
+            {
+                HttpMessageHandler handler = messageHandler ?? new HttpClientHandler();
+                if (StaticSettings.IsDebug)
+                {
+                    handler = new LoggingHandler(handler);
+                }
+
+                functionAppReadyClient = new HttpClient(handler);
+                const string jsonContentType = "application/json";
+                functionAppReadyClient.DefaultRequestHeaders.Add("User-Agent", Constants.CliUserAgent);
+                functionAppReadyClient.DefaultRequestHeaders.Add("Accept", jsonContentType);
+            }
+
+            await RetryHelper.Retry(async () =>
+                  {
+                      functionAppReadyClient.DefaultRequestHeaders.Add("x-ms-request-id", Guid.NewGuid().ToString());
+                      var uri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+                      var request = new HttpRequestMessage()
+                      {
+                          RequestUri = uri,
+                          Method = HttpMethod.Get
+                      };
+
+                      var response = await functionAppReadyClient.SendAsync(request);
+                      ColoredConsole.Write(".");
+
+                      if (response.IsSuccessStatusCode)
+                      {
+                          var json = await response.Content.ReadAsAsync<JToken>();
+                          var processUpTime = json["processUptime"]?.ToObject<int>() ?? 0;
+
+                          // Wait 60sec for the readiness of the worker process rebooting by deployment.
+                          if (processUpTime >= Constants.DefaultWorkerProcessUptimeReadiness)  
+                          {
+                              ColoredConsole.WriteLine(" done");
+                              return;
+                          }
+                          else
+                          {
+                              throw new Exception($"GetFunctions is not ready. Hostname: {functionApp.HostName}");
+                          }
+                      }
+                      else
+                      {
+                          throw new Exception($"Status check returns unsuccessful status. Hostname: {functionApp.HostName}");
+                      }
+                  }, 6, TimeSpan.FromSeconds(10)
+            );
+        }
+
+
+
+    public static async Task PrintFunctionsInfo(Site functionApp, string accessToken, string managementURL, bool showKeys)
+    {
+            ColoredConsole.Write("Waiting for the functions host up and running ");
+            await WaitUntilFunctionAppReadyAsync(functionApp, accessToken, managementURL, showKeys);
```

Comment:
Can we scope the this change to just kubernetes? Otherwise this is a change in behavior for existing apps

Reply:
Done!